colcon_cmake/task/cmake/build.py

```diff
@@ -313,16 +315,26 @@ def _get_make_arguments(self):
 
         if CMAKE_EXECUTABLE is None:
             raise RuntimeError("Could not find 'cmake' executable")
-        cmd = [
-            CMAKE_EXECUTABLE, '--build', args.build_base,
-            '--target', 'install']
+        cmd = [CMAKE_EXECUTABLE]
+        cmake_ver = await get_cmake_version()
+        if cmake_ver and cmake_ver >= parse_version('3.15.0'):
+            # CMake 3.15+ supports invoking `cmake --install [--config]
+            # This only installs, whereas --build <dir> --target install will
+            # build (again) then install.
+            cmd += ['--install', args.build_base]
+        else:
```

Comment:
Why is choosing the `else` block in case the version couldn't be retrieved the right choice?

Reply:
`--install` is a newer feature requiring 3.15 whereas the else statement uses `--build --target` which has had long term support. We can only guarantee support for `--install` if we can validate that the version number is `> 3.15`.

Reply:
I understand that but are you expecting the get version function to fail for any CMake version? If not, please check for the potential returned error and not just silently use the else block.

Reply:
Ok. I'll add an error message in the case wheret eh CMake version cannot be determined.